### PR TITLE
Environment awareness

### PR DESCRIPTION
**Pull request by m4dz**
This PR introduces an environment aware onboarding workflow:
- transmit to client app the environment variables in a dedicated module
- skip the import from google step if the dedicated app isn't available on the cozy
- remove email setup step

### CODE REVIEW

File: server/controllers/authentication.coffee [hunk 1]
```diff
@@ -9,23 +10,52 @@ localization = require '../lib/localization_manager'
 passwordKeys = require '../lib/password_keys'
 
 
+getEnv = (callback) ->
+    async.parallel [
+        (callback) ->
+            User.first (err, user) ->
```
Comment (aenario) [inline server/controllers/authentication.coffee]: I'd make all this into a User.getusername() function.

File: server/controllers/authentication.coffee [hunk 2]
```diff
@@ -9,23 +10,52 @@ localization = require '../lib/localization_manager'
...
+                    username: user.public_name
+                else
+                    username: helpers.hideEmail user.email
+                        .split ' '
+                        .map (word) -> word[0].toUpperCase() + word.slice(1)
+                        .join ' '
+                callback null, env
+
+        , (callback) ->
+            env = apps: (key for key of require('../lib/router').getRoutes())
```
Comment (aenario) [inline server/controllers/authentication.coffee]: use Object.keys

File: server/controllers/authentication.coffee [hunk 3]
```diff
@@ -9,23 +10,52 @@ localization = require '../lib/localization_manager'
...
+            callback null, env
+    ],
+    (err, results) ->
```
Comment (aenario) [inline server/controllers/authentication.coffee]: We don't really need the async.parallel function here, you might as well remove it, we can reintroduce it if we add more env.

File: server/controllers/authentication.coffee (hunk 1 above)
Comment (m4dz) [inline server/controllers/authentication.coffee]: Hmmm, I'm not sure to understand what you mean :confused:

File: server/controllers/authentication.coffee (hunk 2 above)
Comment (m4dz) [inline server/controllers/authentication.coffee]: Of course, damned!

File: server/controllers/authentication.coffee (hunk 3 above)
Comment (m4dz) [inline server/controllers/authentication.coffee]: True. I just wanted to keep things separated to be more flexible, but I agree, I'll rewrite it.

**Pull request merged**